source/time_loop_mgrit.template.h

```diff
@@ -0,0 +1,1127 @@
+//
+// SPDX-License-Identifier: MIT
+// Copyright (C) 2020 - 2023 by the ryujin authors
+//
+
+#pragma once
+
+#include "checkpointing.h"
+#include "introspection.h"
+#include "scope.h"
+#include "solution_transfer.h"
+#include "time_loop_mgrit.h"
+#include "hyperbolic_module.h"
+#include "offline_data.h"
+#include "geometry_cylinder.h"
+#include "discretization.h"
+#include "euler/parabolic_system.h"
+#include "euler/description.h"
+#include "initial_values.h"
+#include "offline_data.h"
+#include "parabolic_module.h"
+#include "postprocessor.h"
+#include "quantities.h"
+#include "time_integrator.h"
+#include "vtu_output.h"
+#include "level_structures.h"
+
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/revision.h>
+#include <deal.II/base/work_stream.h>
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/vector_tools.templates.h>
+
+#include <deal.II/base/point.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/base/symmetric_tensor.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <vector>
+
+#include <fstream>
+#include <iomanip>
+
+using namespace dealii;
+
+namespace ryujin{
+  namespace mgrit
+  {
+
+    template <typename Description, int dim, typename Number>
+    TimeLoopMgrit<Description, dim, Number>::TimeLoopMgrit(const MPI_Comm &mpi_comm,
+                                                           const LevelStructures<Description,dim, Number> &ls,
+                                                           const Number initial_time,
+                                                           const Number final_time)
+    : ParameterAcceptor("/TimeLoop")
+    , mpi_communicator_(mpi_comm)
+    , hyperbolic_system_(ls.hyperbolic_system)
+    , parabolic_system_(ls.parabolic_system)
+    , discretization_(ls.discretization)
+    , offline_data_(ls.offline_data)
+    , initial_values_(ls.initial_values)
+    , hyperbolic_module_(ls.hyperbolic_module)
+    , parabolic_module_(ls.parabolic_module)
+    , time_integrator_(ls.time_integrator)
+    , postprocessor_(ls.postprocessor)
+    , vtu_output_(ls.vtu_output)
+    , quantities_(ls.quantities)
```

Comment:
see above. then you can write all this like
```suggestion
    , level_structures(ls)
```